File: Python/7kyu/Binary scORe.py
# https://www.codewars.com/kata/56cafdabc8cfcc3ad4000a2b/train/python


def score(n):
    if n < 2:
        return n
    i = 0
    while n!=0:
        n>>=1
        i+=1
    return (1<<i)-1


if __name__ == "__main__":
    print(score(0), 0)
    print(score(1), 1)
    print(score(49), 63)
    print(score(1000000), 1048575)

File: C/7kyu/Binary scORe.c
/*
https://www.codewars.com/kata/56cafdabc8cfcc3ad4000a2b/train/c
*/

#include <stdio.h>
#include <limits.h>

unsigned score (unsigned n);


int main(){
    printf("%u", score(49));
    return 0;
}




unsigned score (unsigned n){
  unsigned i = 0;
  unsigned d = n;
  
  while (d!=0){d>>=1;i++;}

  return n==0 ? 0 : n==1 ? 1 : n==UINT_MAX ? UINT_MAX : (1<<i)-1;
  }